ow_sim_tests/scripts/test_arm_move_cartesian_guarded.py

```diff
@@ -0,0 +1,207 @@
+          position = Point(1.86, -0.63, constants.DEFAULT_GROUND_HEIGHT - 0.1),
+          orientation = Quaternion(1, 0.005, 0.053, 0.029)
+        ),
+        force_threshold = 200,
+        torque_threshold = 100
+      ),
+      TEST_NAME, 'test_03_arm_move_cartesian_guarded'
+    )
+
+  def test_04_arm_move_cartesian(self):
+    arm_move_cartesian_result = test_arm_action(self,
+      'ArmMoveCartesian', owl_msgs.msg.ArmMoveCartesianAction,
+      owl_msgs.msg.ArmMoveCartesianGoal(
+        frame = 0,
+        relative = False,
+        pose= Pose(
+          position = Point(1.86, -0.63, constants.DEFAULT_GROUND_HEIGHT + 0.2),
+          orientation = Quaternion(1, 0.005, 0.053, 0.029)
+        ),
+      ),
+      TEST_NAME, 'test_04_arm_move_cartesian'
+    )
+
+  def test_05_arm_move_cartesian(self):
+    arm_move_cartesian_result = test_arm_action(self,
+      'ArmMoveCartesian', owl_msgs.msg.ArmMoveCartesianAction,
+      owl_msgs.msg.ArmMoveCartesianGoal(
+        frame = 0,
+        relative = False,
+        pose= Pose(
+          position = Point(2.263, 0.147, -0.140),
+          orientation = Quaternion(-0.615, 0.626, 0.339, 0.339)
+        ),
+      ),
+      TEST_NAME, 'test_05_arm_move_cartesian'
+    )
+
+  def test_06_arm_move_cartesian_guarded(self):
+    arm_move_cartesian_guarded_result = test_arm_action(self,
+      'ArmMoveCartesianGuarded', owl_msgs.msg.ArmMoveCartesianGuardedAction,
+      owl_msgs.msg.ArmMoveCartesianGuardedGoal(
+        frame = 0,
+        relative = False,
+        pose= Pose(
+          position = Point(2.263, -0.072, -0.157),
+          orientation = Quaternion(0.707, -0.707, 0, 0)
+        ),
+        force_threshold = 200,
+        torque_threshold = 5
+      ),
+      TEST_NAME, 'test_06_arm_move_cartesian_guarded'
+    )
+  
+  def test_07_arm_move_cartesian_guarded(self):
+    arm_move_cartesian_guarded_result = test_arm_action(self,
+      'ArmMoveCartesianGuarded', owl_msgs.msg.ArmMoveCartesianGuardedAction,
+      owl_msgs.msg.ArmMoveCartesianGuardedGoal(
+        frame = 0,
+        relative = False,
+        pose= Pose(
+          position = Point(2.263, -0.072, -0.157),
+          orientation = Quaternion(0.707, -0.707, 0, 0)
+        ),
+        force_threshold = 200,
+        torque_threshold = 50
+      ),
+      TEST_NAME, 'test_07_arm_move_cartesian_guarded'
+    )
+
+  def test_08_arm_move_cartesian_guarded(self):
+    arm_move_cartesian_guarded_result = test_arm_action(self,
+      'ArmMoveCartesianGuarded', owl_msgs.msg.ArmMoveCartesianGuardedAction,
+      owl_msgs.msg.ArmMoveCartesianGuardedGoal(
+        frame = 0,
+        relative = False,
+        pose= Pose(
+          position = Point(1.797, 1.277, -0.160),
+          orientation = Quaternion(0.959, -0.284, 0.003, 0.006)
+        ),
+        force_threshold = 200,
+        torque_threshold = 10
+      ),
+      TEST_NAME, 'test_08_arm_move_cartesian_guarded'
+    )
+
+  def test_09_arm_move_cartesian_guarded(self):
```

Comment:
It looked to me like this test was trying to trigger an torque threshold breach, but that did not happen in my testing. You could try lowering the commanded z-coordinate by a little bit, so that the scoop is pushed more and more into the terrain as it moves, or lower the `torque_threshold` parameter.

Reply:
Ye, not sure why but looks like every time I run the same test, the force and torque are not guaranteed to be the same value, I am going to run some more tests on this.